src/components/FreeRoomsCardList.tsx

```diff
@@ -14,26 +15,26 @@ export const FreeRoomsCardList: FC<{
 }> = ({ searchText, timeSlot, stringDay }) => {
   const rooms = Object.keys(room_occupancy[stringDay]).sort();
 
-  return (
-    <SimpleGrid minChildWidth="340px" spacing="2" mt="4" px={["4", "8"]}>
-      {rooms
-        .filter((room: string) => {
-          const hasRoomName = room.includes(searchText.toUpperCase());
-          const hasAvailableSlot =
-            timeSlot == "All"
-              ? true
-              : room_occupancy[stringDay][room][timeSlot];
+  const fliteredRooms = rooms.filter((room: string) => {
+    const hasRoomName = room.includes(searchText.toUpperCase());
+    const hasAvailableSlot =
+      timeSlot == "All" ? true : room_occupancy[stringDay][room][timeSlot];
+
+    return hasRoomName && hasAvailableSlot;
+  });
 
-          return hasRoomName && hasAvailableSlot;
-        })
-        .map((room: string, i: number) => (
-          <FreeRoomCard
-            key={i}
-            timeSlots={room_occupancy[stringDay][room]}
-            room={room}
-            room_id={room_ids[room]}
-          />
-        ))}
+  return fliteredRooms.length > 0 ? (
```

Comment:
Why don't we extract 

`filteredRoom.length > 0` into a variable so that it now becomes

`return hasResult ? (...) : (...)`

Reply:
ok